src/transformState.js

```
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, obj.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const removeKey of obj.keysToRemove) {
+          delete state[removeKey];
+        }
+
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default : return null;
```

It is better to return an error or a message that something went wrong.